result_test.go

```diff
@@ -408,12 +408,31 @@ func TestToMapBy(t *testing.T) {
 }
 
 func TestToSlice(t *testing.T) {
-	input := []int{1, 2, 3, 4}
-
-	result := []int{}
-	From(input).ToSlice(&result)
+	tests := []struct {
+		input        []int
+		output       []int
+		want         []int
+		wantedOutCap int
+	}{
+		//output is nil slice
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, []int{}, []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 16},
+		//cap(out)>len(result): we get the same slice, just resized.
+		{[]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, make([]int, 0, 11), []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}, 11},
```

Comment:
@cleitonmarx you're not testing the `we get the same slice` part. :) that'd be a pointer comparison.